### ISSUE TITLE
OS X Installation Error

### ISSUE THREAD

**Issue by vicchi** [middleman/middleman#1460]
Possibly a Middleman issue, possibly a State issue which bundles Middleman. I've cheekily opened an issue for Slate as well (see https://github.com/tripit/slate/issues/186), which via copy-and-paste says this ..

Following the (Slate) installation instructions, I get as far as step 5 ... `$ bundle exec middleman server` and then things go horribly wrong.

I'm running with the following environment ...

OS X 10.10.2 (Yosemite)
Ruby `ruby 2.1.5p273 (2014-11-13 revision 48405) [x86_64-darwin14.0]` (installed via Homebrew at `/usr/local/opt/ruby/bin/ruby`)

Repository cloned, `bundle install` run without (visible) errors and then ...

```
$ bundle exec middleman server
/usr/local/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file -- set (LoadError)
	from /usr/local/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /usr/local/gems/bundler-1.8.2/lib/bundler/index.rb:1:in `<top (required)>'
	from /usr/local/gems/bundler-1.8.2/lib/bundler/source/rubygems.rb:74:in `specs'
	from /usr/local/gems/bundler-1.8.2/lib/bundler/source/local_rubygems.rb:7:in `specs'
	from /usr/local/gems/bundler-1.8.2/lib/bundler/lazy_specification.rb:53:in `__materialize__'
	from /usr/local/gems/bundler-1.8.2/lib/bundler/spec_set.rb:88:in `block in materialize'
	from /usr/local/gems/bundler-1.8.2/lib/bundler/spec_set.rb:85:in `map!'
	from /usr/local/gems/bundler-1.8.2/lib/bundler/spec_set.rb:85:in `materialize'
	from /usr/local/gems/bundler-1.8.2/lib/bundler/definition.rb:132:in `specs'
	from /usr/local/gems/bundler-1.8.2/lib/bundler/definition.rb:177:in `specs_for'
	from /usr/local/gems/bundler-1.8.2/lib/bundler/definition.rb:166:in `requested_specs'
	from /usr/local/gems/bundler-1.8.2/lib/bundler/environment.rb:18:in `requested_specs'
	from /usr/local/gems/bundler-1.8.2/lib/bundler/runtime.rb:13:in `setup'
	from /usr/local/gems/bundler-1.8.2/lib/bundler.rb:122:in `setup'
	from /usr/local/gems/bundler-1.8.2/lib/bundler/setup.rb:8:in `<top (required)>'
	from /usr/local/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /usr/local/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:54:in `require'
```

I'm assuming it's an environment issue, but beyond that I'm left scratching my head. Does this ring any bells?

**Comment by tdreyno** [comment]
What does the `Gemfile.lock` look like? Seems weird that it can't find set, which has been in the Ruby core forever

**Comment by vicchi** [comment]
My `Gemfile.lock` is this ...

```
GEM
  remote: http://rubygems.org/
  specs:
    activesupport (4.1.9)
      i18n (~> 0.6, >= 0.6.9)
      json (~> 1.7, >= 1.7.7)
      minitest (~> 5.1)
      thread_safe (~> 0.1)
      tzinfo (~> 1.1)
    celluloid (0.16.0)
      timers (~> 4.0.0)
    chunky_png (1.3.3)
    coffee-script (2.3.0)
      coffee-script-source
      execjs
    coffee-script-source (1.8.0)
    compass (1.0.1)
      chunky_png (~> 1.2)
      compass-core (~> 1.0.1)
      compass-import-once (~> 1.0.5)
      rb-fsevent (>= 0.9.3)
      rb-inotify (>= 0.9)
      sass (>= 3.3.13, < 3.5)
    compass-core (1.0.1)
      multi_json (~> 1.0)
      sass (>= 3.3.0, < 3.5)
    compass-import-once (1.0.5)
      sass (>= 3.2, < 3.5)
    em-websocket (0.5.1)
      eventmachine (>= 0.12.9)
      http_parser.rb (~> 0.6.0)
    erubis (2.7.0)
    eventmachine (1.0.4)
    execjs (2.2.2)
    ffi (1.9.6)
    haml (4.0.6)
      tilt
    hike (1.2.3)
    hitimes (1.2.2)
    hooks (0.4.0)
      uber (~> 0.0.4)
    http_parser.rb (0.6.0)
    i18n (0.6.11)
    json (1.8.2)
    kramdown (1.5.0)
    libv8 (3.16.14.7)
    listen (2.8.5)
      celluloid (>= 0.15.2)
      rb-fsevent (>= 0.9.3)
      rb-inotify (>= 0.9)
    middleman (3.3.7)
      coffee-script (~> 2.2)
      compass (>= 1.0.0, < 2.0.0)
      compass-import-once (= 1.0.5)
      execjs (~> 2.0)
      haml (>= 4.0.5)
      kramdown (~> 1.2)
      middleman-core (= 3.3.7)
      middleman-sprockets (>= 3.1.2)
      sass (>= 3.4.0, < 4.0)
      uglifier (~> 2.5)
    middleman-core (3.3.7)
      activesupport (~> 4.1.0)
      bundler (~> 1.1)
      erubis
      hooks (~> 0.3)
      i18n (~> 0.6.9)
      listen (>= 2.7.9, < 3.0)
      padrino-helpers (~> 0.12.3)
      rack (>= 1.4.5, < 2.0)
      rack-test (~> 0.6.2)
      thor (>= 0.15.2, < 2.0)
      tilt (~> 1.4.1, < 2.0)
    middleman-gh-pages (0.0.3)
      rake (> 0.9.3)
    middleman-livereload (3.3.4)
      em-websocket (~> 0.5.1)
      middleman-core (~> 3.2)
      rack-livereload (~> 0.3.15)
    middleman-sprockets (3.4.1)
      middleman-core (>= 3.3)
      sprockets (~> 2.12.1)
      sprockets-helpers (~> 1.1.0)
      sprockets-sass (~> 1.3.0)
    middleman-syntax (2.0.0)
      middleman-core (~> 3.2)
      rouge (~> 1.0)
    minitest (5.5.1)
    multi_json (1.10.1)
    padrino-helpers (0.12.4)
      i18n (~> 0.6, >= 0.6.7)
      padrino-support (= 0.12.4)
      tilt (~> 1.4.1)
    padrino-support (0.12.4)
      activesupport (>= 3.1)
    rack (1.6.0)
    rack-livereload (0.3.15)
      rack
    rack-test (0.6.3)
      rack (>= 1.0)
    rake (10.4.2)
    rb-fsevent (0.9.4)
    rb-inotify (0.9.5)
      ffi (>= 0.5.0)
    redcarpet (3.2.2)
    ref (1.0.5)
    rouge (1.7.2)
    ruby18_source_location (0.2)
    sass (3.4.9)
    sprockets (2.12.3)
      hike (~> 1.2)
      multi_json (~> 1.0)
      rack (~> 1.0)
      tilt (~> 1.1, != 1.3.0)
    sprockets-helpers (1.1.0)
      sprockets (~> 2.0)
    sprockets-sass (1.3.1)
      sprockets (~> 2.0)
      tilt (~> 1.1)
    therubyracer (0.12.1)
      libv8 (~> 3.16.14.0)
      ref
    thor (0.19.1)
    thread_safe (0.3.4)
    tilt (1.4.1)
    timers (4.0.1)
      hitimes
    tzinfo (1.2.2)
      thread_safe (~> 0.1)
    uber (0.0.13)
    uglifier (2.7.0)
      execjs (>= 0.3.0)
      json (>= 1.8.0)

PLATFORMS
  ruby

DEPENDENCIES
  middleman (~> 3.3.0)
  middleman-gh-pages
  middleman-livereload (~> 3.3.0)
  middleman-syntax
  rake (~> 10.4.0)
  redcarpet (~> 3.2.1)
  rouge (= 1.7.2)
  ruby18_source_location
  therubyracer
  wdm (~> 0.1.0)
```

**Comment by tdreyno** [comment]
Can you update to the latest Middleman, 3.3.9?

Also, `ruby18_source_location`, shouldn't be necessary

**Comment by vicchi** [comment]
No change I'm afraid. I'm assuming something it severely borked in my overall (Homebrew) Ruby environment, so I'm going to move the entire contents of `/usr/local` to somewhere safe and then reinstall Homebrew and Ruby and start from scratch. This might take a while, but I'll report back with progress.

**Comment by vicchi** [comment]
Resolved. I'm not 100% sure what the issue was but it's resolved.

A clean Homebrew install had no problems, so I went back to my original Homebrew environment and Middleman install, removed Middleman, cloned the repo and did a `$ bundle install --path vendor/bundle`, with the same issue occurring.

Removing all locally installed Gems also had no effect.

But ... doing a `$ brew cleanup -n` showed multiple old versions of Ruby installed (but not necessarily active) and after doing a `$ brew cleanup ruby` the problem stopped happening and Middleman now runs just fine (as does Slate).

So it looks like, somehow, my Ruby environment was slightly borked due to the multiple old versions still hanging around. Quite how, I'm not sure. But for now, we're all good.
